### ISSUE TITLE
[Bug] Regression for Biped 1.1 - fails with StackOverflow at Translator.Execute

### ISSUE THREAD

**Issue by TheChatty** [Ryujinx/Ryujinx#4539]
### Description of the issue

Approx. two weeks ago Biped 1.1 was running smoothly. Now it crashes with latest build on MacOS when trying to start the game.
```
00:00:17.778 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:00:18.072 |E| [mvk-info] Created 3 swapchain images with initial size (1920, 1010) and contents scale 1.0 for screen MD 20120.
00:00:18.120 |E| [mvk-info] Created 3 swapchain images with initial size (1920, 1080) and contents scale 1.0 for screen MD 20120.
Stack overflow.
   at ARMeilleure.Translation.Translator.Execute(ARMeilleure.State.ExecutionContext, UInt64)
   at Ryujinx.Cpu.Jit.JitCpuContext.Execute(Ryujinx.Cpu.IExecutionContext, UInt64)
   at Ryujinx.HLE.HOS.ArmProcessContext`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].Execute(Ryujinx.Cpu.IExecutionContext, UInt64)
   at Ryujinx.HLE.HOS.Kernel.Threading.KThread.ThreadStart()
   at System.Threading.Thread+StartHelper.RunWorker()
   at System.Threading.Thread+StartHelper.Run()
   at System.Threading.Thread.StartCallback()
```

At the moment it runs fine Ryujinx 1.0.0 but now and then make the system unstable and emulator also crashes kind of ramdomly.

### Reproduction steps

Install Ryujinx on MacOS Ventura.
Load Biped 1.1 and wait for "Press (A) to start."
Push (A) and notice exception log.

### Log file

[Ryujinx_1.1.0+3ac8e0f_2023-03-12_15-23-28.log](https://github.com/Ryujinx/Ryujinx/files/10951095/Ryujinx_1.1.0%2B3ac8e0f_2023-03-12_15-23-28.log)

### OS

MacOS Ventura 13.2

### Ryujinx version

1.1.0+3ac8e0f

### Game version

1.1.0

### CPU

M2

### GPU

M2

### RAM

16GB

### List of applied mods

n/a

### Additional context?

Going back in Ryujinx builds doesn't help. Exception is due to an external lib probably.

**Comment by gdkchan** [comment]
Are you testing with or without the hypervisor? Because the log you attached indicates the hypervisor is enabled, yet the exception is on the JIT. Those stack overflows are usually caused by the fact that the default stack size on macOS is very low (it is less than 1 MB). You can avoid them by increasing the stack size for the guest threads, for example by changing this line https://github.com/Ryujinx/Ryujinx/blob/master/Ryujinx.HLE/HOS/Kernel/Threading/KThread.cs#L192 to `HostThread = new Thread(ThreadStart, 0x100000);`.

**Comment by TheChatty** [comment]
How can I switch using/avoiding hypervisor? I'll try the higher stack size...

**Comment by gdkchan** [comment]
I think you are probably running a x64 build that needs to use Rosetta emulation to run (and can't use hypervisor). You should use a arm64 build. I think CI does not produce arm64 builds right now, so you will probably need to build it yourself if you want to use the latest version.

> Changing between latest build and old official version makes my loose all my settings. Shall I create a new issue for this?

I'm not sure if that can be considered a bug. Migrating from a newer configuration version to a older one is not something that we support right now.

**Comment by TheChatty** [comment]
```
% file Ryujinx*
Ryujinx:                        Mach-O 64-bit executable x86_64
Ryujinx.Ava:                    Mach-O 64-bit executable x86_64
```

**Comment by TheChatty** [comment]
I just compiled HEAD myself. HV was activated. No longer crashes. So it might be the JIT?

**Comment by gdkchan** [comment]
As I explained above, code executed using the JIT is limited to the stack size of the thread where it is running. The hypervisor has no such limitation as the stack memory is managed by the emulator and has the size that the game requests. On top of that, it is running the game code directly, so the stack usage will be the same as actual Switch hardware. JIT generated code stack usage is usually higher as the code generated is different from the original and less efficient.
